client/src/controllers/InitController.test.js

```diff
@@ -148,4 +148,39 @@ describe('InitController', () => {
       ]);
     });
```

Comment:
We also want to add a test for the ability to call `preventDefault` and stop the dispatching of custom events.

I recommend we add something like the following test at around line 151. Within the `describe('when using custom event names', () => {` block.

```js
    it('should support the ability to block the additional events and classes removal', async () => {
      jest.clearAllMocks();

      expect(handleEvent).not.toHaveBeenCalled();

      document.addEventListener(
        'w-init:ready',
        (event) => {
          event.preventDefault();
        },
        { once: true },
      );

      const article = document.createElement('article');
      article.id = 'article';
      article.innerHTML = `<p data-controller="w-init" data-w-init-event-value="custom:event">CONTENT</p>`;

      document.body.append(article);

      await jest.runAllTimersAsync();

      // only once - the w-init, no other events should fire
      expect(handleEvent).toHaveBeenCalledTimes(1);
    });
```

Reply:
Hey, @lb- 
Please do correct me if I am wrong, but there's only one `eventListener` here, so even if another event could be called, there wouldn't be a way to actually know. Should I add a second eventListener for the `custom:event`? So that we'd be sure that the `handleEvent` gets called only once because of the logic, and not the fact that it was only listening for one thing.

Reply:
You could add another assertion that the event type is what you expect. 

Something like

```js
expect(handleEvent).toHaveBeenCalledTimes(1);
expect(handleEvent).toHaveBeenLastCalledWith(expect.objectContaining({ type: "w-init:ready" }));
```

If you look further up near the top of the unit test, we already have an event listener for `custom:event`.

Reply:
Yep. See 
https://github.com/wagtail/wagtail/blob/5e122f764dcedb33b73b549f981be3e5b96f1427/client/src/controllers/InitController.test.js#L110

Reply:
AH, yes! 
Thank you!